tracardi/process_engine/action/v1/converters/base64/decode/plugin.py

```diff
@@ -0,0 +1,79 @@
+from base64 import b64decode
+
+from tracardi.service.plugin.domain.register import Form, FormGroup, FormField, FormComponent, \
+    Documentation, PortDoc
+from tracardi.service.plugin.domain.register import Plugin, Spec, MetaData
+from tracardi.service.plugin.domain.result import Result
+from tracardi.service.plugin.runner import ActionRunner
+from .model.models import Configuration
+
+
+def validate(config: dict):
+    return Configuration(**config)
+
+
+class Base64DecodeAction(ActionRunner):
+    config: Configuration
+
+    async def set_up(self, init):
+        self.config = validate(init)
+
+    async def run(self, payload: dict, in_edge=None):
+        dot = self._get_dot_accessor(payload)
+        base64 = dot[self.config.from_base64]
+        result = b64decode(base64 + '==').decode(self.config.output_encoding)
+
+        return Result(port="payload", value={"text": result})
+
+
+def register() -> Plugin:
+    return Plugin(
+        start=False,
+        spec=Spec(
+            module=__name__,
+            className=Base64DecodeAction.__name__,
+            inputs=['payload'],
+            outputs=['payload'],
+            version='0.1',
+            license='MIT',
+            author='knittl',
+            manual='base64_decode',
+            init={
+                'from_base64': 'payload@base64',
+                'output_encoding': 'utf-8'
+            },
+            form=Form(groups=[
+                FormGroup(
+                    fields=[
+                        FormField(
+                            id='from_base64',
+                            name='Base64-encoded data to decode to string',
+                            description='Type path to string or string itself. Default source is payload.',
+                            component=FormComponent(type='dotPath', props={'defaultSourceValue': 'payload'})
+                        ),
+                        FormField(
+                            id='output_encoding',
+                            name='Encoding',
+                            description='Output text encoding. Will be used to convert output bytes to text to bytes '
+                                        'after base64-decoding it. Set to "utf-8" if unsure.',
+                            component=FormComponent(type='text', props={'label': 'encoding'})
```

Comment:
You could use different FormComponent to make the selection of encoding simpler for the user, e.g.

 FormComponent(type='select', props={'label': 'encoding', "items": {
"utf-8": "UTF-8",
"ascii": "ASCII",
...
}})

Reply:
I thought about it and I tried to use this. But the [list of standard encodings](https://docs.python.org/3.10/library/codecs.html#standard-encodings) is long. I don't want to to add all of those to the select component. It doesn't seem it is possible to have a select component with freeform input?

A compromise I could think of: provide the most common encodings via select component and if a user needs a custom/non-default encoding, they must go to the raw JSON config directly. What do you think?

Reply:
@knittl that is a good approach.

Reply:
Great, then I will implement a select with the 5 or 10 most-common encodings :)